SpatialGDK/Source/SpatialGDKEditor/Private/SchemaGenerator/SpatialGDKEditorSchemaGenerator.cpp

```diff
@@ -68,6 +68,13 @@
 			return false;                                                                                                                  \
 		}                                                                                                                                  \
 	} while (false)
+
+#define COND_SCHEMA_GEN_ERROR_AND_RETURN(Condition, Format, ...)                                                                                           \
```

Comment:
My main worry with this is that it's obscuring that it returns on this. And it doesn't save that much code because most of it's in the log message.

Reply:
The other question I'd have is what errors we are trying to detect. It looks like it's mostly reporting things that would have caused schema compilation to fail. Although that said we might at some point move to directly writing schema bundles at which point these kinds of errors can happen.

Reply:
Yeah, Mike commented something similar, which is why I added the `_AND_RETURN` at the end, so it's clear that this potentially alters execution flow.

Turns out the pattern here was really common, because we don't want to crash on schema errors, nor do we want to proceed with parsing the schema file, so had to make a re-usable thing which did those. The macro seemed like the easiest way forward.